content/src/content/jcr_root/apps/core/wcm/components/modal/v1/modal/clientlibs/site/js/modal.js

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ * Copyright 2019 Adobe Systems Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+/* Getting Page URL  Modal ID */
+
+var getValOnPage = document.createElement("div");
+getValOnPage.setAttribute("id", "data-modal-content");
+document.body.appendChild(getValOnPage);
+var getUrlModalID;
+
+(function() {
+    "use strict";
+
+    function getModalOpenBehavior() {
+        var getUrlModal = window.location.hash.substr(1).split("?")[0];
+        var isModalOn = document.querySelectorAll("div[data-modal-show='true']");
+        if ((getUrlModal !== "" && isModalOn.length === 0) || (getUrlModal !== "" && isModalOn.length >= 0)) {
+            return (
+                getUrlModalID = getUrlModal
+            );
+        } else if ((getUrlModal === "" && isModalOn.length > 0)) {
+            var isModalOnVal = isModalOn[0].id;
+            return (
+                getUrlModalID = isModalOnVal
+            );
+        } else if ((getUrlModal !== "" && isModalOn.length === 0)) {
+            return (
+                getUrlModalID = getUrlModal
+            );
+        }
+    }
+
+    function fetchData(url, insertModalContent) {
+        var xhttp;
```

Comment:
what is the reason for requesting the content client side and not rendering the xf / cf server side?

Reply:
@richardhand XF and CF are reusable content. If we render it server side, then it will get cached within the modal component. Is it desirable?
With the Client side, we can always get the updated content while it is still cached on the dispatcher.